src/2d_nonlocal_async.cpp

```diff
@@ -9,23 +9,93 @@
 #include <hpx/hpx_init.hpp>
 #include <hpx/hpx.hpp>
 
+#include <hpx/include/parallel_algorithm.hpp>
+#include <boost/range/irange.hpp>
+
 #include <cstddef>
 #include <cstdint>
 #include <iostream>
 #include <vector>
 #include <math.h>
-#include <algorithm> 
+#include <unordered_map>
+#include <algorithm>
+#include <fstream>
 
-#include "point.h"
-#include "print_time_results.hpp"
+#include "../include/point.h"
+#include "../include/print_time_results.hpp"
+#include "../include/writer.h"
 
 //mathematical constant PI
 #define PI 3.14159265
```

Comment:
Please use MP_I from the cmath header, instead of defining your own constant.

Reply:
Ok done. Please check latest commit